Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +463,190 @@ def restore_snapshot_online(
         snapshot_name,
         {})
 
+def create_schedule(
+        client_obj,
+        storage_system_ip,        
+        storage_system_username,
+        storage_system_password,
+        schedule_name,
+        snapshot_name,
+        base_volume_name,
+        read_only,
+        expiration_time,
+        retention_time,
+        expiration_unit,
+        retention_unit,
+        task_freq, 
+        task_freq_custom):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule creation failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Schedule name is null",
+            {})                 
+    if snapshot_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Snapshot name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule creation failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+
+    if len(snapshot_name) < 1 or len(snapshot_name) > 20:
+        return (False, False, "Schedule create failed. Snapshot name must be atleast 1 character and not more than 20 characters", {})
+    if base_volume_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Base volume name is null",
+            {})
+    if len(base_volume_name) < 1 or len(base_volume_name) > 31:
+        return (False, False, "Schedule create failed. Base volume name must be atleast 1 character and not more than 31 characters", {})    
+
+    expirationHours = convert_to_hours(expiration_time, expiration_unit)
+    retentionHours = convert_to_hours(retention_time, retention_unit)
+
+    freq = "@hourly"
+    if retentionHours > expirationHours:
+       return (False, False, "Expiration time must be greater than or equal to retention time", {})   
+
+    if task_freq_custom and task_freq:
+       return (False, False, "Provide either task_freq or task_freq_custom", {})        
+
+    if task_freq_custom:          
+       if ' ' in task_freq_custom:            
+           task_custom_list = str(task_freq_custom).split()
+           if len(task_custom_list) == 5:
+               minutes_task = task_custom_list[0]               
+               hour_task = task_custom_list[1]
+               day_task = task_custom_list[2]
+               month_task = task_custom_list[3]  
+               day_of_week_task = task_custom_list[4]
+               if '*' not in minutes_task:
+                  if (int(minutes_task) > 50 or int(minutes_task) < 0):
```

Comment:
Why is reason behind checking for minuts>50?

Reply:
minute should be between 0-50